xbmc/filesystem/File.h

```diff
@@ -103,11 +104,11 @@ class CFile
 
   static int DetermineChunkSize(const int srcChunkSize, const int reqChunkSize);
 
-  BitstreamStats* GetBitstreamStats() { return m_bitStreamStats; }
+  BitstreamStats* GetBitstreamStats() { return m_bitStreamStats.get(); }
```

Comment:
I'm thinking that calling `.get()` conflicts with the idea of a wrapped pointer. We can probably return a `std::unique_ptr&`, and let the caller make the decision to unwrap only if needed.

Going further, would could return a `const std::unique_ptr&` and label this function const to improve const-correctness. This might be a new commit or PR if propagating const-correctness requires changes in calling code.

Reply:
I considered going do the path of returning a `unique_ptr` but thought it would be too much bloat for this PR. I'll look into it again and see how much work it may be. If it's trivial I'll include the changes in this PR otherwise I can make a follow up PR.

Reply:
Returning a `unique_ptr&` should be trivial - find all compiler errors an add a `.get()`. Adding const-correctness is more work, and do whatever is easiest for you (this PR or a new one) - I can review either way

Reply:
updated, it was trivial